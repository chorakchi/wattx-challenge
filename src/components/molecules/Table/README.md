Table component display information in a table view format of rows and columns. 

<iframe width= "100%" height= "600px" src="https://www.figma.com/embed?embed_host=astra&url=https://www.figma.com/file/2LokuyVBjWW20P4CcDJGra/Top-Coins?node-id=7%3A1144" ></iframe>
Not: Design and implementation has some differentiate. 



### APIs
this component had props for heading and items. and it accepting array for them.
as wel as noResultsLabel props for the time component doesn't have item as input  to show a message instead of empty table.

### Playground:
