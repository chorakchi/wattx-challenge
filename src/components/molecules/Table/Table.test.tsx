import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Table } from "./Table";
import { mockData } from "./Table.mock";

jest.mock("components/atoms/SVG/SVG", () => ({
  SVG: "svg",
}));

describe("Table", () => {

  
  it("should render the Table", () => {

    const subject = shallow(<Table {...mockData} />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {

    const subject = renderer.create(<Table {...mockData} />).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
