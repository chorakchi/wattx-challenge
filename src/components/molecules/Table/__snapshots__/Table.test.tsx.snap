// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render correctly and match to the Snapshot 1`] = `
<div
  className="Boxstyled__BoxStyled-sc-12nhtob-0 bOPHwF Tablestyled__StyledWrapper-sc-1bwnru4-1 dVBaTC"
>
  <table
    className="Tablestyled__StyledTable-sc-1bwnru4-0 bLIjVx"
  >
    <thead>
      <tr>
        <td
          className="Tablestyled__TextStyled-sc-1bwnru4-2 Tablestyled__StyledHeading-sc-1bwnru4-3 cbPEmA czeCIj"
        >
          <span
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm"
          >
            
          </span>
        </td>
        <td
          className="Tablestyled__TextStyled-sc-1bwnru4-2 Tablestyled__StyledHeading-sc-1bwnru4-3 cbPEmA czeCIj"
        >
          <span
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm"
          >
            Name
          </span>
        </td>
        <td
          className="Tablestyled__TextStyled-sc-1bwnru4-2 Tablestyled__StyledHeading-sc-1bwnru4-3 cbPEmA czeCIj"
        >
          <span
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm"
          >
            Price
          </span>
        </td>
        <td
          className="Tablestyled__TextStyled-sc-1bwnru4-2 Tablestyled__StyledHeading-sc-1bwnru4-3 cbPEmA czeCIj"
        >
          <span
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm"
          >
            Price Change (24h)
          </span>
        </td>
        <td
          className="Tablestyled__TextStyled-sc-1bwnru4-2 Tablestyled__StyledHeading-sc-1bwnru4-3 cbPEmA czeCIj"
        >
          <span
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm"
          >
            Market Cap
          </span>
        </td>
        <td
          className="Tablestyled__TextStyled-sc-1bwnru4-2 Tablestyled__StyledHeading-sc-1bwnru4-3 cbPEmA czeCIj"
        >
          <span
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm"
          >
            Volume (24h)
          </span>
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTIiIGhlaWdodD0iNTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              className="icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                className="icon"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCBTC.svg&w=128&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCBTC.svg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCBTC.svg&w=128&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            Bitcoin
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            60789.66
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 iHigjS"
          >
            -1.43
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            1145611846639
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            38412572337
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTIiIGhlaWdodD0iNTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              className="icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                className="icon"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCETH.svg&w=128&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCETH.svg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCETH.svg&w=128&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            Ethereum
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            3865.11
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 iHigjS"
          >
            -0.61
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            455901921929
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            17834027180
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTIiIGhlaWdodD0iNTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              className="icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                className="icon"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCBNB.svg&w=128&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCBNB.svg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCBNB.svg&w=128&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            Binance Coin
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            466.48
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 iHigjS"
          >
            -1.10
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            78432909179
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            2070636046
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTIiIGhlaWdodD0iNTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              className="icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                className="icon"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCADA.svg&w=128&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCADA.svg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCADA.svg&w=128&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            Cardano
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            2.19
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 iHigjS"
          >
            -1.50
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            71956827935
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            2605292904
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTIiIGhlaWdodD0iNTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              className="icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                className="icon"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCUSDT.svg&w=128&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCUSDT.svg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCUSDT.svg&w=128&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            Tether
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            1.00
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            0.00
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            68753034847
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            70584860663
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTIiIGhlaWdodD0iNTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              className="icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                className="icon"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCXRP.svg&w=128&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCXRP.svg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCXRP.svg&w=128&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            XRP
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            1.14
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 iHigjS"
          >
            -0.05
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            53640103617
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            4051590232
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTIiIGhlaWdodD0iNTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              className="icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                className="icon"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCSOL.svg&w=128&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCSOL.svg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fs3-symbol-logo.tradingview.com%2Fcrypto%2FXTVCSOL.svg&w=128&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            Solana
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            158.70
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 iHigjS"
          >
            -2.53
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            47654350081
          </div>
        </td>
        <td>
          <div
            className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm Tablestyled__TextStyled-sc-1bwnru4-2 cbPEmA"
          >
            1946537911
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
